app/webpack/shared/components/text_editor.jsx

```diff
@@ -34,30 +35,31 @@ class TextEditor extends React.Component {
 
   render( ) {
     const {
-      content,
       maxLength,
       placeholder,
+      changeHandler,
       className,
       textareaClassName,
       showCharsRemainingAt,
       onBlur
     } = this.props;
-    const { textareaChars, preview } = this.state;
-    let textareaOnChange;
-    if ( maxLength ) {
-      textareaOnChange = e => {
+    const { textareaChars, preview, content } = this.state;
+    const textareaOnChange = e => {
+      if ( changeHandler ) { changeHandler( e.target.value ); }
+      this.setState( { content: e.target.value } );
+      if ( maxLength ) {
         if ( e.target.value.length > showCharsRemainingAt ) {
           this.setState( { textareaChars: e.target.value.length } );
         }
-      };
-    }
+      }
+    };
     return (
       <div className={`TextEditor ${className} ${preview && "with-preview"}`}>
-        { this.textarea && this.textarea.current && (
+        { this.textarea && (
```

Comment:
Hm, why isn't `current` necessary? Don't you need to be sure the ref is pointing at an actual DOM element?

Reply:
So I guess I had the understanding that with a ref, the actual DOM element represented by `.current` is only truly set after the render but before the `componentDidMount` in the component's lifecycle. What I thought that meant was that `.current` is best used in functions outside of render such as: `onClick={

Reply:
I moved this check to the handle click function of the TextEditorFormatButton itself, still operating under the assumption that refs are set after render. I also passing a function to setState of the TextEditor's content as I noticed that simply clicking the button without any other changes (for example selecting text and clicking bold or putting in a template/placeholder text) wasn't updating the content correctly.